devito/types/basic.py

```diff
@@ -724,9 +725,20 @@ def dimensions(self):
         """Tuple of Dimensions representing the object indices."""
         return self._dimensions
 
+    @property
+    def _eval_deriv(self):
+        return self
+
+    @property
```

Comment:
cached as it's got cost > 0

Reply:
Well that'd defeat the all purpose of it caching it as `f(x)` and `f(x+dx/2)` would both return True instead of True and False.

Reply:
but those are different objects, and you're caching on the instance, not on the class

also: `self.indices, self.indices_ref, self.dimensions` they all are immutable (as expected), so the output of this property is deterministic and immutable too